plugin-modernizer-cli/src/main/java/io/jenkins/tools/pluginmodernizer/cli/Main.java

```diff
@@ -66,8 +72,25 @@ public String getVersion() {
         }
     }
 
+    public void listAvailableRecipes() {
+        Yaml yaml = new Yaml();
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream("recipe_data.yaml");
+        if (inputStream == null) {
+            LOG.error("Failed to load the available recipes.");
+            return;
+        }
+        Map<String, Object> data = yaml.load(inputStream);
+        LOG.info("Available recipes:");
+        Map<String, Map<String, String>> recipes = (Map<String, Map<String, String>>) data.get("recipes");
```

Comment:
Would be nice to have some kind of model for this. Using such raw type make the code difficult to read

For example 

```Java
List<Recipe> recipes = (List<Recipe>) data.get("recipes");
```

Is much more easy to understand.

So if the format change we can change the model directly and not dig into the code to find map of map of string

Can you try to introduce a new class that will represent the Java structure of a recipe in `recipe_data.yaml` ?

I think also the type of object is not ideal inside `recipe_data.yaml` A list of recipe would be better that an map of recipes. What do you think ?

Reply:
Understood. Let me create a model.